caseta_to_mqtt/caseta/button_watcher.py

```diff
@@ -0,0 +1,225 @@
+                )
+            if button_state.value == ButtonState.NOT_PRESSED:
+                self.tracking_started_at = datetime.now()
+            button_state.value = button_state.value.next_state()
+
+    @property
+    def is_timed_out(self) -> bool:
+        return (
+            self.tracking_started_at is not None
+            and (datetime.now() - self.tracking_started_at)
+            > BUTTON_WATCHER_MAX_DURATION
+        )
+
+
+class ButtonWatcher:
+    def __init__(
+        self,
+        remote: PicoRemote,
+        button_id: ButtonId,
+        event_handler: EventHandler,
+    ):
+        self._remote: PicoRemote = remote
+        self._button_id: ButtonId = button_id
+        self._event_handler: EventHandler = event_handler
+        self.button_history: ButtonHistory = ButtonHistory()
+
+    async def button_watcher_loop(self) -> None:
+        button_history = self.button_history
+
+        button_log_prefix = (
+            f"remote: <id: {self._remote.device_id}, "
+            f"name: {self._remote.name}>, "
+            f"button:{self._button_id}"
+        )
+
+        button_tracking_window_end = datetime.now() + BUTTON_WATCHER_MAX_DURATION
+        await asyncio.sleep(DOUBLE_CLICK_WINDOW.total_seconds())
+        async with button_history.button_state.get() as locked_current_state:
+            current_state = locked_current_state.value
+            if current_state == ButtonState.FIRST_PRESS_AND_FIRST_RELEASE:
+                LOGGER.debug(f"{button_log_prefix}: A single press has completed")
+                button_history.is_finished = True
+                await self._event_handler.handle_event(
+                    CasetaEvent(
+                        self._remote,
+                        self._button_id,
+                        ButtonEvent.SINGLE_PRESS_COMPLETED,
+                    )
+                )
+                return
+            elif current_state == ButtonState.FIRST_PRESS_AWAITING_RELEASE:
+                LOGGER.debug(
+                    f"{button_log_prefix}: a long press has started but not finished"
+                )
+                await self._event_handler.handle_event(
+                    CasetaEvent(
+                        self._remote,
+                        self._button_id,
+                        ButtonEvent.LONG_PRESS_ONGOING,
+                    )
+                )
+            elif current_state == ButtonState.DOUBLE_PRESS_FINISHED:
+                LOGGER.debug(f"{button_log_prefix}: A double press has completed")
+                button_history.is_finished = True
+                await self._event_handler.handle_event(
+                    CasetaEvent(
+                        self._remote,
+                        self._button_id,
+                        ButtonEvent.DOUBLE_PRESS_FINISHED,
+                    )
+                )
+                return
+            else:
+                LOGGER.debug(f"{button_log_prefix}: current state is {current_state}")
```

Comment:
```suggestion
                LOGGER.debug("%s: current state is %s", button_log_prefix, current_state)
```

python logging utilities still like %s templated strings